### PR TITLE
[MM-49378] Added Nginx SLO for Enterprise customer

### PR DESCRIPTION
**Pull request by mirshahriar**
#### Summary

We need to provide list of enterprise group while running provisioner.

```bash
$ cloud server [flags]

Flags:
      --enterprise-groups strings.       The list of enterprise group ids to create dedicated Nginx SLOs for.
```

Sample SLO
```yaml
objective: 99.9
sli:
  events:
    errorQuery: sum(rate(nginx_ingress_controller_request_duration_seconds_count{exported_service='mm-d5wh',status=~'(5..|429|499)'}[{{.window}}]))
    totalQuery: sum(rate(nginx_ingress_controller_request_duration_seconds_count{exported_service='mm-d5wh'}[{{.window}}]))
```

#### Ticket Link

  Fixes [MM-49378](https://mattermost.atlassian.net/browse/MM-49378)

#### Release Note

```release-note
Added Nginx SLO for enterprise customer
```

### CODE REVIEW

File: internal/provisioner/nginx_sli.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+	return sli
+}
+
+func createOrUpdateNginxSLIs(clusterInstallation *model.ClusterInstallation, k8sClient *k8s.KubeClient, logger log.FieldLogger) error {
+	sli := makeNginxSLIs(clusterInstallation)
+	return createOrUpdateClusterPrometheusServiceLevel(sli, k8sClient, logger)
+}
+
+func deleteNginxSLI(clusterInstallation *model.ClusterInstallation, k8sClient *k8s.KubeClient, logger log.FieldLogger) error {
```
Comment (fmartingr) [inline internal/provisioner/nginx_sli.go:68]: Can we use something like `ensureInstallationNginxSLIDeleted` to easily understand that the function won't `err` if the SLI doesn't exist?

**Review by fmartingr — APPROVED** [review]
👍

File: cmd/cloud/server_flag.go [hunk 2]
```diff
@@ -90,6 +91,7 @@ func (flags *provisioningParams) addFlags(command *cobra.Command) {
 	command.Flags().StringVar(&flags.s3StateStore, "state-store", "dev.cloud.mattermost.com", "The S3 bucket used to store cluster state.")
 	command.Flags().StringSliceVar(&flags.allowListCIDRRange, "allow-list-cidr-range", []string{"0.0.0.0/0"}, "The list of CIDRs to allow communication with the private ingress.")
 	command.Flags().StringSliceVar(&flags.sloInstallationGroups, "slo-installation-groups", []string{}, "The list of installation group ids to create dedicated SLOs for.")
+	command.Flags().StringSliceVar(&flags.enterpriseGroups, "enterprise-groups", []string{}, "The list of enterprise group ids to create dedicated Nginx SLOs for.")
```
Comment (gabrieljackson) [inline cmd/cloud/server_flag.go:94]: nit: let's use `slo-enterprise-groups` so that it matches the other slo flag above

**Review by gabrieljackson — COMMENTED** [review]
Looks good. Just one small request.

**Comment by stafot** [comment]
Please change default objective to 99.5

**Review by gabrieljackson — APPROVED** [review]
LGTM!

**Pull request merged**